internal/pkg/aws/acm/acm.go

```diff
@@ -66,23 +70,32 @@ func (a *ACM) ValidateCertAliases(aliases []string, certs []string) error {
 	}
 	for _, alias := range aliases {
 		if !validAliases[alias] {
-			return fmt.Errorf("%s is not a valid domain against %s", alias, strings.Join(certs, ","))
+			return &errInValidAliasAgainstCert{
+				certs:         certs,
+				alias:         alias,
+				domainsOfCert: domainsOfCert,
+			}
 		}
 	}
 	return nil
 }
 
-func (a *ACM) findValidAliasesAgainstCert(ctx context.Context, aliases []string, cert string) ([]string, error) {
+func (a *ACM) validDomainsOfCert(ctx context.Context, cert string) ([]string, error) {
 	resp, err := a.client.DescribeCertificateWithContext(ctx, &acm.DescribeCertificateInput{
 		CertificateArn: aws.String(cert),
 	})
 	if err != nil {
 		return nil, fmt.Errorf("describe certificate %s: %w", cert, err)
 	}
-	domainSet := make(map[string]bool)
-	domainSet[aws.StringValue(resp.Certificate.DomainName)] = true
-	for _, san := range resp.Certificate.SubjectAlternativeNames {
-		domainSet[aws.StringValue(san)] = true
+	var domainsOfCert []*string
+	domainsOfCert = append(domainsOfCert, resp.Certificate.SubjectAlternativeNames...)
```

Comment:
Are we sure that we are also including the `DomainName` as well that we are getting in the resp? Because when I looked at the tests I got little confused that we have removed `DomainName` entry and added it's value inside `SubjectAlternativeNames`. It should be good as long as `SubjectAlternativeNames` has DomainName already included in it. Else we will lose one entry every time.

Reply:
From this [doc](https://docs.aws.amazon.com/acm/latest/APIReference/API_CertificateDetail.html#ACM-Type-CertificateDetail-SubjectAlternativeNames) it is mentioned that:
The subject alternative names include the canonical domain name (CN) of the certificate and additional domain names that can be used to connect to the website. 
So the `DomainName` is already included in the `SubjectAlternativeNames`. That is the reason why I removed the `DomainName` and  added to the `SubjectAlternativeNames` in the test cases.